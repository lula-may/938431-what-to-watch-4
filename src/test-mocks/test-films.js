const movieTitles = [
  `Jurassic Park`,
  `Gone With The Wind`,
  `The Godfather`,
  `Casablanca`,
  `A Streetcar Named Desire`,
  `Some Likes It Hot`,
];

const actors = [
  [`Sam Neill`, `Jeff GoldBlum`, `Laura Dern`],
  [`Tom Cruise`, `Rayan Gosling`],
  [`Anne Hathaway`, `Nicole Kidman`, `Keanu Reeves`],
  [`Will Smith`, `Bruce Willis`, `Mila Yovovich`],
  [`Sandra Bullock`, `Daniel Craig`],
  [`Robert De Niro`, `Vin Diesel`, `Charlize Theron`]
];

const reviews = [
  {
    author: `Betty Boughter`,
    date: new Date(2019, 6, 15, 23, 50),
    id: `Betty `,
    rating: `6.6`,
    text: `Betty Boughter bought some butter.`
  },
  {
    author: `Jack Daniels`,
    date: new Date(2020, 5, 2, 23, 50),
    id: `jack`,
    rating: `7.7`,
    text: `Jack and Jill went up the hill.`
  },
  {
    author: `Robert Burns`,
    date: new Date(2020, 2, 2, 23, 50),
    id: `burns`,
    rating: `8.8`,
    text: `Should old acquaintance be forgot and never brought to mind.`
  }
];

const MAX_RATING = 9.7;
const RATING_COUNT = 150;
const MAX_RELEASE_YEAR = 2000;

const directors = [`Stiven Spielberg`, `Mr. Good`, `Mr. Bad`, `Mrs. Ugly`, `Ms. Red`, `Mr. Black`];

const genres = [`Sci-Fi`, `War`, `Detective`, `Fantasy`, `Sport`, `Adventure`];

const testMovies = movieTitles.map((title, i) => {
  const rating = (MAX_RATING - i).toFixed(1);
  return {
    actors: actors[i],
    bigPoster: `bip-poster-${i}.jpg`,
    src: `video-${i}.mp3`,
    description: {
      paragraphs: [`bla-bla-bla`, `olya-lya-lya`, `wow-wow`]
    },
    director: directors[i],
    genre: genres[i],
    id: i.toString(),
    poster: `poster-${i}.jpg`,
    rating: {
      count: RATING_COUNT,
      score: rating,
    },
    releaseYear: MAX_RELEASE_YEAR - i,
    reviews,
    runTime: 60 + i,
    title,
  };
});


export {testMovies};

