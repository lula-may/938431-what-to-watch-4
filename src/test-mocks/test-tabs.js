const tabs = [
  {
    id: `aa`,
    type: `First`
  },
  {
    id: `bb`,
    type: `Second`,
  },
  {
    id: `cc`,
    type: `Third`,
  }
];

export {tabs};
