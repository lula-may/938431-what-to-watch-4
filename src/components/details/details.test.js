import React from "react";
import renderer from "react-test-renderer";
import Details from "./details.jsx";
import {testMovies} from "../../test-mocks/test-films";

const movie = testMovies[0];

it(`should render correctly tab Details component`, () => {
  const tree = renderer.create(
      <Details movie={movie}/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
