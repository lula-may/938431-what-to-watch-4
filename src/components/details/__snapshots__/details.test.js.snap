// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly tab Details component 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Stiven Spielberg
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        Sam Neill,
        <br />
        Jeff GoldBlum,
        <br />
        Laura Dern
        <br />
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        1h 0m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Sci-Fi
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2000
      </span>
    </p>
  </div>
</div>
`;
